### PR TITLE
Make [edd_restrict] check if License Expired

### PR DESCRIPTION
**Pull request by mintplugins**
Make the edd_restrict shortcode check if a license is expired before displaying content.

### CODE REVIEW

File: extensions/software-licensing/content-restriction-compatibility.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+				foreach ( $licenses as $license_post ) {
+					
+					$license_key = get_post_meta( $license_post->ID, '_edd_sl_key', true );
+					
+					foreach( $download_ids as $download_id ){
+						if ( $sl->is_download_id_valid_for_license( $download_id['download'], $license_key ) ){
+							
+							$license_status = $sl->get_license_status( $license_post->ID );
+														
+							if ( $license_status != 'expired' ){
```
Comment (pippinsplugins) [inline extensions/software-licensing/content-restriction-compatibility.php]: This should also check that the license has not been revoked.

**Comment by grappler** [comment]
This is the code that I am using. I am not sure if it is better, just shorter.

```php
/*
 * Check if user has a valid subscription
 */
function wpzoo_edd_is_active_customer( $user_id = '' ) {
	// Fail if EDD Software Licencing not active
	if ( ! function_exists('edd_software_licensing') ) {
		return;
	}
	$license_keys = edd_software_licensing()->get_license_keys_of_user( $user_id );
	foreach ( $license_keys as $license ) {
		$status = edd_software_licensing()->get_license_status( $license->ID );
		if ( 'expired' !== $status ) {
			return true;
		}
	}
}
```

My code has the same problem checking if the licences is published or not. I suppose a better function would be `edd_software_licensing()->check_license()`.

**Comment by mintplugins** [comment]
@grappler Thanks for chiming in! :) However yours only checks if they have ANY active license. In this snippet here, we are checking it against a specific download.

**Comment by grappler** [comment]
ah!, yes :smile:

**Comment by pippinsplugins** [comment]
Thanks @mintplugins !

**Pull request merged**